src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics_generator/linux/LinuxMountedPartitionMetricsGenerator.java

```diff
@@ -0,0 +1,48 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics_generator.linux;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.MountedPartitionMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.hwnet.MountedPartitions;
```

Comment:
This depends on MountedPartitions, but MountedPartitions also depends on LinuxMountedPartitionMetricsGenerator. Is this ok? Can't you move the sampling logic from MountedPartitions into the addSample() function here?

Reply:
So the generator calls addSample() in MountedPartitions class, where it updates its own map and sets it back in the generator. The idea is to have all logic related to profiling/measuring these metrics in one class separate from the generator. The other hardware metrics generators follow the same pattern as well. Didn't want to diverge as well.

> The logic in here is good but does the ES process always have access to /proc/mounts? This collector may fail in certain ODFE distributions.

In the distros we support, /proc/mounts has 777 access. So reading them should be fine.